index.js

```diff
@@ -53,11 +56,59 @@ var InvalidJoinSourceError = TypedError({
     actual: null
 });
 
+var AppRequiredError = TypedError({
+    type: 'ringpop.options-app.required',
+    message: 'Expected `options.app` to be a non-empty string.\n' +
+        'Must specify an app for ringpop to work.\n'
+});
+
+var OptionsRequiredError = TypedError({
+    type: 'ringpop.options.required',
+    message: 'Expected `options` argument to be passed.\n' +
+        'Must specify options for `RingPop({ ... })`.\n'
+});
+
+var HostPortRequiredError = TypedError({
+    type: 'ringpop.options-host-port.required',
+    message: 'Expected `options.hostPort` to be valid.\n' +
+        'Got {hostPort} which is not {reason}.\n' +
+        'Must specify a HOST:PORT string.\n',
+    hostPort: null,
+    reason: null
+});
+
 function RingPop(options) {
     if (!(this instanceof RingPop)) {
         return new RingPop(options);
     }
 
+    if (!options) {
+        throw OptionsRequiredError();
+    }
+
+    if (typeof options.app !== 'string' ||
+        options.app.length === 0
+    ) {
+        throw AppRequiredError();
+    }
+
+    var isString = typeof options.hostPort === 'string';
+    var parts = options.hostPort && options.hostPort.split(':');
+    var isColonSeperated = parts && parts.length === 2;
+    var isIP = parts && parts[0] && parts[0].match(IP_PATTERN);
```

Comment:
asserting on an IP is going to fuck people up. current apps use hostname for host:port